### ISSUE TITLE
Feature Request: Different home screens for different contexts

### ISSUE THREAD

**Issue by qall** [sduduzog/slim-launcher#14]
First of all thank you for this great app.

Related to Issue #12 , I would suggest a middle way (because an app search would be a backward feature, you said in your app presentation that changing the app on the homescreen is part of the mechanism):

a selection  of apps based on **contexts**;

## Work:
- App 1 (e.g. Mail)
- App 2 (e.g. Calc)
- App 3 
- App 4
- App 5

## Transports:
- App 6 (e.g. Maps)
- App 7 (e.g. Podcasts)
- App 1
- App 3
- App 8

And so on.

Contexts could be limited to 3, and accessed with swipes

With this:
- we don't scroll on our app drawer with the app search (=> distraction)
- we have a more personalized set of apps, depending on contexts

This is similar to Ubuntu mobile OS design, which I find very compelling.

Thanks for your review!

**Comment by ReleaseTheKnowledge** [comment]
I agree with this. As im using different apps depending on context. And also cheating a bit with 5 apps on Samsung "apps edge". With this I could remove it completely.

But still I think that we need to keep in mind that the goal of the launcher is to have less apps, and only have the essential. And if we add too much then it just becomes any other launcher.

An idea is to add settings menu for different levels of minimalism. Maybe a "Zen mode" which is bare minimum allowed. And also a less strict mode where this is allowed. Then it would feel OK. And it can give some progression as you can strive to go to a more strict level as times goes on.

**Comment by uzluisf** [comment]
This sounds like a great idea, @qall. But, as noted by @ReleaseTheKnowledge, this'll likely clutter up the app. My proposal is to implement app search so as to keep with the launcher's goal: A static wallpaper-less homescreen with up to five apps. Thus no horizontal swipes and only accessing the launcher's settings by swiping up.

>  we don't scroll on our app drawer with the app search (=> distraction)

Ideally, you wouldn't scroll up and the app search would pop up. I envision just a field (similar to **Notes**, **Preferences**, etc.) with the word **Search** (or something similar) that I'd tap and take me to the list of currently installed apps. **Search** doesn't need to be enabled by default. I don't understand how it'll be distracting since it'll be disabled by default and placed in the *Slim options*  if enabled.

**Comment by qall** [comment]
Thank you for your answers @ReleaseTheKnowledge and @uzluisf 

> I envision just a field (similar to **Notes**, **Preferences**, etc.) with the word **Search** (or something similar) that I'd tap and take me to the list of currently installed apps. **Search** doesn't need to be enabled by default. I don't understand how it'll be distracting since it'll be disabled by default and placed in the _Slim options_ if enabled.

I agree with that. It's more in line with the spirit of the app.

Cheers and keep up the great work!

**Comment by uzluisf** [comment]
@qall You shouldn't have closed your issue. I was just voicing my opinion :-) and ultimately it's up to @sduduzog what they think it's best. I think your feature is a great idea if implemented as an option.

**Comment by ReleaseTheKnowledge** [comment]
Yeah, I think it's a good idea! I would actually prefer @qall's idea over a straight up search, as there is already a "search" in the launcher (but a slow one, which is good) . Having to place apps on a homescreen makes you think about what apps are essential. And an option for having multiple home screen would make someone having >5 "essential" apps not having to replace apps as much and at the same time keep the apps in an organized list.

I think the app should encourage using less apps. And an app search does in my option encourage using more apps than having 2-3 carefully picked homepages.

In my above post I was just voicing my concern that making the launcher too easy to use may hurt its goal. As I know it's hard to not "cheat" (being effective) when given the option.

As i said in my other post I think it's a good idea to have different "levels" you can choose from in the options, that will determine what settings is visible and allowed. So for example if search have been added, and you are at the "zen" level/mode (or whatever you wanna call it), then having to change your level before you can enable search would make it a bit less tempting to quickly enable search, search the app and then disable it (or forget to disable it). Instead of doing the longer version of replacing the apps on your homepage.

There should be something that discourages enabling too many options and encourages using less. And adding levels would accomplish that while making it a bit fun like a game, and still allowing for some customization for those who have not quite reached the zen mode of using their apps (like me).

Something like this.
Zen mode:
* 3 apps allowed.
* "change launcher" is not visible

Normal mode:
* 5 apps allowed
+ Option if "change launcher is visible"

"Easier" mode:
+ Option if either search or "multiple home screens" should be used

"Easy" mode:
+ all customization is allowed.

**Comment by sduduzog** [comment]
@ReleaseTheKnowledge I like how you layed this out. I won't lie, I am still in a phase of finding the perfect identity of this app. Play sore reviews and emails I'm receiving are binary. Some want more customization options and some want the app to be more strict. Going back to why it was built in the first place, a tool to a minimalist lifestyle, and as such, people choose to do it in different ways, hence the varying opinions. Following this model will cater directly for this need. I'm just concerned in how long it would take me. I'm job hunting so I have a few hours a week to work on this.

**Comment by ReleaseTheKnowledge** [comment]
@sduduzog alright, no stress. It's all up to you if you have the time. If you want you could implement it incrementally. Start with one of the features. And slowly build more on top.
And thank you for the great launcher!

**Comment by Mikaela** [comment]
I am not sure where to comment, but I have a problem that it's just impossible for me to fit my app usage to 7 apps and I am not sure if contexts could help me (without having tried them) or if three would be enough). I will try to document my usage here in case it's any help either towards planning contextes or you can give me pointers on what to abandon.

At the moment I have picked: 

* Meditation Assistant - I should meditate more often.
* Down Dog - I should Yoga more
* RadioDroid - I have oversensitive sense of hearing and I try to work with it in some public places by having something else coming to my ears.
* Habitica - Daily tasks / todo list
* Keep (notes) - grocery list, I need quick access to daily
* Wire - family
* Signal - some friends and support people
* + I am cheating by having KISS Launcher as assist app.

What gets closed out, but I would find important to have listed somehow more or less often:

* 112 Suomi - 112 is emergency number in European Union and the app would transmit coordinates from GPS to the emergency services more accurately than triangulating in case I wasn't sure where I was.
* \<app stores to update apps at times in case autoupdate didn't work, this includes Aurora/Play Store and F-Droid\>
* \<2FA apps\> - because I mainly use Bitwarden, but some force me to use Authy and Steam doesn't allow me to get the 2FA secret out of it without having a specific rooted phone
* Conversations - XMPP client that I use with my partners and where my IRC usage and a lot else has moved to. It however has persistent notification, so it may not be necessary to have it in launcher and it can be better to access it difficultly as I overuse it.
* \<Web browser\> - I tend to open them through cheated KISS Launcher search though.
* \<mail\> - which I would be happier only checking on computer though.
* TrustedContacts, CircleOfSix, Life360 - I have been harassed and while it's not a daily event, I would feel safer with these emergency apps more easily reachable so people I trust would know where I am and possibly call me pretending to need me or help me calm after the event.
* Lock - as the only other way to lock the screen is pressing power button.
* Maps - which goes under navigation
* Media Volume Overlay - as this phone has too loud minimum volume level
* Messages - because some people still use SMSes

There are also a lot more apps that I access seldom or only in specific circumstances such as traveling abroad, but I tried to list only those that I find necessary on my phone. I am not using Facebook (Messenger/WhatsApp/Instagram), Telegram or Discord [but am using too many other apps](https://mikaela.info/#im), so people get divided under more apps than they might for another person.

Looking at my list, I wonder am I more addicted to my phone than everyone else or is Slim Launcher just not for me?

PS. I also didn't realize that clock is shortcut to clock at first and only when I am typing this I realized that the reason I cannot add a Calendar shortcut is that touching the date opens the Calendar for me.

**Comment by sduduzog** [comment]
@Mikaela You use so many apps!!. But reading through the list I would also have a hard time eliminating. Someone recommended KISS Launcher to me a while back and I loved it. It is a great cheat.

The issue with this though doesn't have one correct way of solving.  I do hope though things get better over time.

**Comment by PEsteves8** [comment]
What about this approach:

The 7 app limit remains, but there's an extra slot called "Temporary App Slot". If the user needs to use an app that isn't one of the main 7 apps - and it's natural that this might happen occasionally - the user can go to 'options' and tap "Add temporary app" and chose one for the search. This adds an 8th slot for 60 seconds that disappears after the time is up or if the users opens it.

It would be a way of keeping a low limit of apps, but explicitly allowing users to open other apps without having to remove a current app and replacing it. Surely many people already use the 7th slot as a replaceable one.

Maybe this wouldn't really solve anyone's problem :D but I feel some users would appreciate having "permission" to open other apps, while keeping in mind that whichever app they're opening, they're doing it as an exception.

If anything else, maybe someone reading this comment can think of some other idea based on this one :)

**Comment by supperino** [comment]
I think a good implementation is to use the side slides for some apps.
Like, swipe left one app, swipe down search engine or browser, and swipe right another app.

**Comment by jurf** [comment]
Just quickly popping in, I haven’t read the whole issue but I saw this wasn’t mentioned, which might interest you: https://experiments.withgoogle.com/morph. :-)

**Comment by sduduzog** [comment]
@jurf That is really awesome. That basically sums up what I'll be leaning towards but context switching will be manual and/or time-based depending on the user preference.

**Comment by jurf** [comment]
I personally also like the location-based feature, because at school I use a completely different app set than at home or when out-of-house.
